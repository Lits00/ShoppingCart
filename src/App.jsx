import Container from './components/Display/Container';
import Footer from './components/Footer';
import Header from './components/Header/Header';
import { useState } from 'react';
import CartContext from './components/Context/cartContext';

function App() {
  const [cart, setCart] = useState([]);

  return (
    <CartContext.Provider value={{ cart, setCart }}>
      <>
        <Header />
        <Container />
        <Footer />
      </>
    </CartContext.Provider>
  );
}

export default App;
