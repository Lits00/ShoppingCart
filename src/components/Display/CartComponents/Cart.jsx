import { useContext } from 'react';
import CartContext from '../../Context/cartContext';
import CartCard from './CartCard';
import '../style/cart.css';
import Total from './Total';

export default function Cart() {
  const { cart, setCart } = useContext(CartContext);

  const removeToCart = (id) => {
    const newCart = cart.filter((item) => item.id !== id);
    setCart(newCart);
  };

  return (
    <section data-testid="cartSection" className="cartPage">
      <div className="cartItems">
        {cart.map((item) => (
          <CartCard
            key={item.id}
            image={item.image}
            name={item.name}
            price={item.price}
            quantity={item.quantity}
            handleClick={() => removeToCart(item.id)}
          />
        ))}
        <Total />
      </div>
    </section>
  );
}
