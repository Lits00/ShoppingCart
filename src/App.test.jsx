import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import App from './App';
import Cart from './components/Display/CartComponents/Cart';
import CartContext from './components/Context/CartContext';

// vitest with react testing library
// test suite (describe) contains all related tests
describe('Initial render of App component', () => {
  // test cases (it/test)
  it('renders home page', () => {
    const { container } = render(<App />, { wrapper: BrowserRouter });
    // console.log(container.outerHTML);

    // assertion (expect().toMatch etc.)
    expect(container).toMatchSnapshot();
  });

  //   it('renders nav buttons', () => {
  // render(<App />, { wrapper: BrowserRouter });
  // console.log(screen.getByRole('navigation').outerHTML);

  // expect(screen.getByRole('navigation')).toBeInTheDocument();
  //   });

  //   it('renders main container', () => {
  // render(<App />, { wrapper: BrowserRouter });
  // console.log(screen.getByRole('main').outerHTML);
  // expect(screen.getByRole('main')).toBeInTheDocument();
  //   });
});

describe('Nav bar links', () => {
  it('redirects to home page', async () => {
    const user = userEvent.setup();

    render(<App />, { wrapper: BrowserRouter });
    const homeBtn = screen.getByRole('link', { name: 'Home' });

    await user.click(homeBtn);
    expect(window.location.pathname).toEqual('/');
  });

  it('redirects to shop page', async () => {
    const user = userEvent.setup();

    render(<App />, { wrapper: BrowserRouter });
    const shopBtn = screen.getByRole('link', { name: 'Shop' });

    await user.click(shopBtn);
    expect(window.location.pathname).toEqual('/shop');
  });

  it('redirects to cart page', async () => {
    const user = userEvent.setup();

    render(<App />, { wrapper: BrowserRouter });
    const cartBtn = screen.getByTestId('ShoppingBasketIcon');

    await user.click(cartBtn);
    expect(window.location.pathname).toEqual('/cart');
  });
});

describe('Cart page render', () => {
  it('renders initial cart page', () => {
    const cart = [];
    const setCart = () => {};

    render(
      <CartContext.Provider value={{ cart, setCart }}>
        <Cart />
      </CartContext.Provider>,
      { wrapper: BrowserRouter }
    );
    expect(screen.getByTestId('cartSection')).toBeInTheDocument();
    expect(screen.getByTestId('cartRedirect')).toBeInTheDocument();
  });
});
