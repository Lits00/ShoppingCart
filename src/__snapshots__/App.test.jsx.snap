// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initial render of App component > renders home page 1`] = `
<div>
  <header>
    <nav>
      <ul>
        <li>
          <a
            class="btn undefined"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class="btn undefined"
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class="cartBtn"
            href="/cart"
          >
            <span
              class="MuiBadge-root css-1c32n2y-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeMedium css-11pbyhm-MuiSvgIcon-root"
                data-testid="ShoppingBasketIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-zkqb44-MuiBadge-badge"
              />
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <section
      class="homepage"
    >
      <div
        class="homeLogo"
      >
        <img
          alt="homepage-logo"
          src="./homelogo.svg"
        />
        <p>
          <a
            class="attribute"
            href="https://www.freepik.com/free-vector/ecommerce-web-page-concept-illustration_21727022.htm#query=e-commerce&position=2&from_view=search&track=sph"
          >
            Image by storyset
          </a>
           
          on Freepik
        </p>
      </div>
      <div>
        <h1>
          Online Store
        </h1>
        <p>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Unde, officiis repudiandae! Dolores odio minus quos ea nulla inventore provident velit beatae. Repellendus quas deleniti culpa suscipit dolore fuga totam quae!
        </p>
        <a
          class="btn defaultBtn"
          href="/shop"
        >
          Shop Now
        </a>
      </div>
    </section>
  </main>
  <footer>
    <p>
      Built by Lits
    </p>
  </footer>
</div>
`;
